enzyme/Enzyme/TypeAnalysis/TypeAnalysis.h

```diff
@@ -174,6 +174,16 @@ class TypeResults {
   /// The TypeTree of a particular Value
   TypeTree query(llvm::Value *val) const;
 
+  /// Whether any part of the top level register can contain a float
+  ///   e.g. { i64, float } can contain a  float, but { i64, i8* } would not.
+  //    Of course, here we compute with type analysis rather than llvm type
+  bool anyFloat(llvm::Value *val) const;
+
+  /// Whether any part of the top level register can contain a pointer
+  ///   e.g. { i64, i8* } can contain a pointer, but { i64, float } would not.
+  //    Of course, here we compute with type analysis rather than llvm type
+  bool anyPointer(llvm::Value *val) const;
+
```

Comment:
I would prefer a naming scheme along the lines of `isAnyFloat`, `isAnyPointer` or `isAnyFloatType`, `isAnyPointerType` or `isFloat`, `isPointer`